services/core-api/app/api/projects/project/resources/project.py

```diff
@@ -72,3 +81,211 @@ def get(self):
+            else:
+                id = project.major_mine_application.major_mine_application_id
+                guid = project.major_mine_application.major_mine_application_guid
+                status_code = project.major_mine_application.status_code
+
+            status_code_filter_pass = update_timestamp_filter_pass = False
+
+            if args['status_code'] and status_code in args['status_code'] or args['status_code'] is None:
+                status_code_filter_pass = True
+
+            update_timestamp_args = args['update_timestamp']
+
+            if args['update_timestamp']:
+                update_timestamp_args = datetime.strptime(args['update_timestamp'], "%Y-%m-%d")
+                update_timestamp_filter = update_timestamp_args.replace(tzinfo = pytz.UTC)
+                update_timestamp = update_timestamp_filter.replace(tzinfo = pytz.UTC)
+
+                if update_timestamp_filter >= update_timestamp :
+                    update_timestamp_filter_pass = True
+            else:
+                update_timestamp_filter_pass = True
+
+            record=None
+            if status_code_filter_pass and update_timestamp_filter_pass:
+                record ={
+                    'stage': stage,
+                    'id':id,
+                    'guid': guid,
+                    'project_title': project.project_title,
+                    'project_id': project.project_id,
+                    'project_guid': project.project_guid,
+                    'mrc_review_required': project.mrc_review_required,
+                    'status_code': status_code,
+                    'contacts': project.contacts,
+                    'update_timestamp': update_timestamp,
+                    'mine': project.mine
+                }
+
+            if record:
+                most_recent_projects.append(record)
+
+
+        return {
+            'records': most_recent_projects,
+            'current_page': pagination_details.page_number,
+            'total_pages': pagination_details.num_pages,
+            'items_per_page': pagination_details.page_size,
+            'total': pagination_details.total_results,
+        }
+
+
+    @classmethod
+    def _build_filter(cls, model, field, op, argfield):
+        return {'model': model, 'field': field, 'op': op, 'value': argfield}
+
+
+    @classmethod
+    def _build_spec(cls, model, fields):
+        return {'model': model, 'fields': fields}
+
+
+    def _apply_filters_and_pagination(self, args):
+        sort_models = {
+            'project_title': 'Project',
+            'project_id': 'Project',
+            'mine_commodity_code': 'MineTypeDetail',
+            'status_code':'ProjectSummary',
+            'status_code':'InformationRequirementsTable',
+            'status_code': 'MajorMineApplication',
+            'update_timestamp': 'ProjectSummary',
+            'update_timestamp': 'InformationRequirementsTable',
+            'update_timestamp': 'InformationRequirementsTable'
```

Comment:
I think this third one was meant to be MajorMineApplication?

Reply:
That's correct. Thank you for noticing